File: Java/Queue/Queue.java
package queue;

public class Queue {

    Node head, tail;

    public Queue() {
        head = null;
        tail = null;
    }

    boolean isEmpty() {
        boolean flag = head == null;
        return flag;
    }

    void add(Node node) {
        if (isEmpty()) {
            head = node;
            tail = node;
        } else {
            head.next = node;
            head = node;
        }
    }

    Node del() {
        if (isEmpty()) {
            System.out.println("Queue is empty !");
            return null;
        } else {
            Node deleted = tail;
            tail = tail.next;
            return deleted;
        }
    }

    void show() {
        if (isEmpty()) {
            System.out.println("Queue is empty !");
        } else {
            Node temp = tail;

            while (temp != null) {
                System.out.println(temp.number);
                temp = temp.next;
            }
        }
    }
}

File: C-plus-plus/Queue/queue.cpp
//
// Created by berkan on 1.09.2021.
//

#include "queue.h"
#include <iostream>

Queue::Queue()
{
    Queue::head = nullptr;
    Queue::final = nullptr;
}

bool Queue::isEmpty()
{
    if (Queue::head == nullptr) return true;
    return false;
}

void Queue::add(Node *node)
{
    if (isEmpty())
    {
        Queue::head = node;
        Queue::final = node;
    } else
    {
        Queue::final->next = node;
        Queue::final = node;
    }
}

int Queue::del()
{
    if (isEmpty())
    {
        std::cout << "Queue is empty" << std::endl;
    } else
    {
        Node* tmp = Queue::head;
        int deleted = tmp->number;
        Queue::head = Queue::head->next;
        delete tmp;
        return deleted;
    }
}

void Queue::show()
{
    Node * tmp = Queue::head;

    while (tmp != nullptr)
    {
        std::cout << tmp->number << std::endl;
        tmp = tmp->next;
    }
}